File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/asynctasks/RankingsLoader.java
package com.devingotaswitch.rankings.asynctasks;

import android.os.AsyncTask;
import android.util.Log;

import androidx.appcompat.app.AlertDialog;

import com.devingotaswitch.fileio.LocalSettingsHelper;
import com.devingotaswitch.fileio.RankingsDBWrapper;
import com.devingotaswitch.rankings.RankingsHome;
import com.devingotaswitch.rankings.domain.DailyProjection;
import com.devingotaswitch.rankings.domain.Draft;
import com.devingotaswitch.rankings.domain.LeagueSettings;
import com.devingotaswitch.rankings.domain.Player;
import com.devingotaswitch.rankings.domain.Rankings;
import com.devingotaswitch.rankings.domain.Team;
import com.devingotaswitch.utils.GeneralUtils;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import java.util.List;
import java.util.Map;

public class RankingsLoader {
    private static final String TAG = "RankingLoader";

    public class RanksLoader extends AsyncTask<Object, String, Rankings> {
        private final AlertDialog pdia;
        private final RankingsHome act;
        private long start;
        private final RankingsDBWrapper rankingsDB;

        public RanksLoader(RankingsHome activity, RankingsDBWrapper rankingsDB) {
            this.pdia = new MaterialAlertDialogBuilder(activity)
                .setCancelable(false)
                .setTitle("Please wait")
                .setMessage("Loading the rankings...")
                .create();
            this.act = activity;
            this.rankingsDB = rankingsDB;
        }

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
            pdia.show();
        }

        @Override
        protected void onPostExecute(Rankings result) {
            super.onPostExecute(result);
            pdia.dismiss();
            Log.d(TAG, GeneralUtils.getLatency(start) + " to load from file");
            act.processNewRankings(result, false);
        }

        @Override
        protected Rankings doInBackground(Object... data) {
            start = System.currentTimeMillis();
            LeagueSettings currentLeague = rankingsDB.getLeague(act, LocalSettingsHelper.getCurrentLeagueName(act));
            Map<String, Player> players = rankingsDB.getPlayers(act);
            Map<String, Team> teams = rankingsDB.getTeams(act);
            List<String> orderedIds = rankingsDB.getPlayersSorted(act, currentLeague);
            Draft draft = LocalSettingsHelper.loadDraft(act, currentLeague.getTeamCount(), currentLeague.getAuctionBudget(),
                    currentLeague.getName(), players);
            Map<String, List<DailyProjection>> playerProjectionHistory = rankingsDB.getPlayerProjectionHistory(act);
            return Rankings.init(teams, players, orderedIds, currentLeague, draft, playerProjectionHistory);
        }
    }

    public class RanksSaver extends AsyncTask<Object, String, Void> {
        private final AlertDialog pdia;
        private final RankingsHome act;
        private long start;
        private final RankingsDBWrapper rankingsDB;

        public RanksSaver(RankingsHome activity, RankingsDBWrapper rankingsDB) {
            this.pdia = new MaterialAlertDialogBuilder(activity)
                .setCancelable(false)
                .setTitle("Please wait")
                .setMessage("Saving the rankings...")
                .create();
            this.act = activity;
            this.rankingsDB = rankingsDB;
        }

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
            pdia.show();
        }

        @Override
        protected void onPostExecute(Void result) {
            super.onPostExecute(result);
            pdia.dismiss();
            Log.d(TAG, GeneralUtils.getLatency(start) + " to save to file");
        }

        @Override
        protected Void doInBackground(Object... data) {
            start = System.currentTimeMillis();
            Map<String, Player> players = (Map<String, Player>) data[0];
            Map<String, Team> teams = (Map<String, Team>) data[1];
            rankingsDB.saveTeams(act, teams.values());
            rankingsDB.savePlayers(act, players.values());
            return null;
        }
    }
}

File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/asynctasks/RankingsLoader.kt
package com.devingotaswitch.rankings.asynctasks

import android.os.AsyncTask
import android.util.Log
import androidx.appcompat.app.AlertDialog
import com.devingotaswitch.fileio.LocalSettingsHelper
import com.devingotaswitch.fileio.RankingsDBWrapper
import com.devingotaswitch.rankings.RankingsHome
import com.devingotaswitch.rankings.domain.DailyProjection
import com.devingotaswitch.rankings.domain.Player
import com.devingotaswitch.rankings.domain.Rankings
import com.devingotaswitch.rankings.domain.Team
import com.devingotaswitch.utils.GeneralUtils.getLatency
import com.google.android.material.dialog.MaterialAlertDialogBuilder

class RankingsLoader {

    class RanksLoader(activity: RankingsHome, private val rankingsDB: RankingsDBWrapper) : AsyncTask<Any?, String?, Rankings?>() {
        private val pdia: AlertDialog = MaterialAlertDialogBuilder(activity)
                .setCancelable(false)
                .setTitle("Please wait")
                .setMessage("Loading the rankings...")
                .create()
        private val act: RankingsHome = activity
        private var start: Long = 0
        override fun onPreExecute() {
            super.onPreExecute()
            pdia.show()
        }

        override fun onPostExecute(result: Rankings?) {
            super.onPostExecute(result)
            pdia.dismiss()
            Log.d(TAG, getLatency(start).toString() + " to load from file")
            act.processNewRankings(result, false)
        }

        override fun doInBackground(vararg data: Any?): Rankings? {
            start = System.currentTimeMillis()
            val userLeagues = rankingsDB.getLeagues(act)
            val players = rankingsDB.getPlayers(act)
            val teams = rankingsDB.getTeams(act)
            val currentLeague = userLeagues.currentLeague
            val orderedIds = rankingsDB.getPlayersSorted(act, currentLeague)
            val draft = LocalSettingsHelper.loadDraft(act, currentLeague.teamCount, currentLeague.auctionBudget,
                    currentLeague.name, players)
            val playerProjectionHistory: MutableMap<String, MutableList<DailyProjection>> =
                    rankingsDB.getPlayerProjectionHistory(act)
            return Rankings.init(teams, players, orderedIds, userLeagues, draft, playerProjectionHistory)
        }
    }

    class LeaguesLoader(activity: RankingsHome, private val rankingsDB: RankingsDBWrapper) : AsyncTask<Any?, String?, Rankings?>() {
        private val pdia: AlertDialog = MaterialAlertDialogBuilder(activity)
                .setCancelable(false)
                .setTitle("Please wait")
                .setMessage("Loading the leagues...")
                .create()
        private val act: RankingsHome = activity
        private var start: Long = 0
        override fun onPreExecute() {
            super.onPreExecute()
            pdia.show()
        }

        override fun onPostExecute(result: Rankings?) {
            super.onPostExecute(result)
            pdia.dismiss()
            Log.d(TAG, getLatency(start).toString() + " to load from file")
            act.setLeague(result)
        }

        override fun doInBackground(vararg data: Any?): Rankings? {
            start = System.currentTimeMillis()
            val userLeagues = rankingsDB.getLeagues(act)
            return Rankings.initWithDefaults(userLeagues)
        }
    }

    class RanksSaver(activity: RankingsHome, private val rankingsDB: RankingsDBWrapper) : AsyncTask<Any?, String?, Void?>() {
        private val pdia: AlertDialog = MaterialAlertDialogBuilder(activity)
                .setCancelable(false)
                .setTitle("Please wait")
                .setMessage("Saving the rankings...")
                .create()
        private val act: RankingsHome = activity
        private var start: Long = 0
        override fun onPreExecute() {
            super.onPreExecute()
            pdia.show()
        }

        override fun onPostExecute(result: Void?) {
            super.onPostExecute(result)
            pdia.dismiss()
            Log.d(TAG, getLatency(start).toString() + " to save to file")
        }

        override fun doInBackground(vararg data: Any?): Void? {
            start = System.currentTimeMillis()
            val players = data[0] as Map<String, Player?>
            val teams = data[1] as Map<String, Team?>
            rankingsDB.saveTeams(act, teams.values)
            rankingsDB.savePlayers(act, players.values)
            return null
        }
    }

    companion object {
        private const val TAG = "RankingsLoader"
    }
}